components/datetime/src/date.rs

```diff
@@ -75,10 +81,38 @@ pub trait IsoTimeInput {
     fn fraction(&self) -> Option<FractionalSecond>;
 }
 
+/// Inputs relevant to formatting a time zone.
+///
+/// Only the GMT offset is required, since it is the final format fallback.
+///
+/// All data represented in TimeZoneInput should be locale-agnostic.
+pub trait TimeZoneInput {
+    /// The GMT offset in Nanoseconds.
+    fn gmt_offset(&self) -> GmtOffset;
+
+    /// The IANA TimeZone identifier.
+    /// TODO(#606) switch this to BCP-47 identifier.
+    fn time_zone_id(&self) -> Option<&str>;
+
+    /// The MetaZone identifier.
+    /// TODO(#528) switch to a compact, stable ID.
+    fn metazone_id(&self) -> Option<&str>;
+
+    /// The time variant (e.g. "daylight", "standard")
+    fn time_variant(&self) -> Option<&str>;
+
+    /// A country code associated with this time zone.
+    fn country_code(&self) -> Option<&str>;
```

Comment:
There are a couple instances in the more complex fallbacking described in UTS-35 that use the country code, for example:

> In composing the metazone + city or country:
>
>    Use the fallbackFormat "{1} ({0})", where:
>        {1} will be the metazone
>        {0} will be a qualifier (city or country)
>
>    If the localized country name is not available, use the code:
>        CU (country code)→ "CU" // no localized country name for Cuba

I'm going to remove this for now. I don't want to commit to adding `country_code()` to the public trait definition just yet, because that might not be the right place for it.

I've filed an issue for this: #620